### DISCUSSION TITLE
Multiple unicast routing table support

### DISCUSSION THREAD

**Discussion by mnxx** [troglobit/pimd#223] [Q&A] [answered]
Hi,

I am testing a policy-based unicast routing scheme which generates a different routing table for each policy. Is there currently any possibility to choose a specific unicast table to use with pimd?

Thank you in advance!

**Comment by troglobit** [comment, marked as answer]
Hi!  There's nothing today, although most multicast routing is concerned with reverse-path filtering (RPF) it's often centered around the source IP of the sender, and how to reach that (to make multicast routing decisions).  So it's well suited for at least source routing.

Just by googling a bit I found https://olegkutkov.me/2019/03/24/getting-linux-routing-table-using-netlink/ which discusses how to look up the correct table. So modifying `src/netlink.c` shouldn't be too much work. Only issue is how this should be configured, and here I'd have a look at cisco/juniper/procurve and also FRR to see what they've done.

**Reply by mnxx** [reply]
Thanks, the provided link is quite insightful!
